inc/cartridge.class.php

```diff
@@ -623,109 +621,100 @@ static function getStatus($date_use, $date_out) {
     * @return boolean|void
    **/
    static function showForCartridgeItem(CartridgeItem $cartitem, $show_old = 0) {
-      global $DB, $CFG_GLPI;
+      global $DB;
 
       $tID = $cartitem->getField('id');
       if (!$cartitem->can($tID, READ)) {
          return false;
       }
       $canedit = $cartitem->can($tID, UPDATE);
 
-      $query = "SELECT COUNT(*) AS count
-                FROM `glpi_cartridges`
-                WHERE (`cartridgeitems_id` = '$tID')";
-
-      if ($result = $DB->query($query)) {
-         $total  = $DB->result($result, 0, "count");
+      if (!$show_old) { // NEW
+         $where = " AND `glpi_cartridges`.`date_out` IS NULL";
+      } else { //OLD
+         $where = " AND `glpi_cartridges`.`date_out` IS NOT NULL";
+      }
 
-         if (!$show_old) { // NEW
-            $where = " AND `glpi_cartridges`.`date_out` IS NULL";
-         } else { //OLD
-            $where = " AND `glpi_cartridges`.`date_out` IS NOT NULL";
-         }
+      $stock_time       = 0;
+      $use_time         = 0;
+      $pages_printed    = 0;
+      $nb_pages_printed = 0;
+      $ORDER = " `glpi_cartridges`.`date_use` ASC,
+                 `glpi_cartridges`.`date_out` DESC,
+                 `glpi_cartridges`.`date_in`";
 
-         $stock_time       = 0;
-         $use_time         = 0;
-         $pages_printed    = 0;
-         $nb_pages_printed = 0;
-         $ORDER = " `glpi_cartridges`.`date_use` ASC,
-                    `glpi_cartridges`.`date_out` DESC,
+      if (!$show_old) {
+         $ORDER = " `glpi_cartridges`.`date_out` ASC,
+                    `glpi_cartridges`.`date_use` ASC,
                     `glpi_cartridges`.`date_in`";
+      }
+      $query = "SELECT `glpi_cartridges`.*,
```

Comment:
Why not switching to iterator?

Reply:
I was only focusing on errors/warnings detected by code analysis tools.